Slim/Menu/BrowseLibrary/Releases.pm

```diff
@@ -72,15 +82,28 @@ sub _releases {
 	my %releaseTypes;
 	my %contributions;
 	my %isPrimaryArtist;
+	my %albumList;
 	foreach (@$albums) {
+		my $genreMatch = undef;
+		if ( $prefs->get('useComposerAsArtist')==2 ) {
+			my $requestRef = [ 'genres', 0, MAX_ALBUMS, "album_id:".$_->{id} ];
+			my $request = Slim::Control::Request->new( $client ? $client->id() : undef, $requestRef );
+			$request->execute();
+			$log->error($request->getStatusText()) if $request->isStatusError();
+			my $genres= $request->getResult('genres_loop');
+			foreach my $genre (@$genres) {
+				$genreMatch = grep {uc($genre->{genre}) eq $_} split(/,/, uc($prefs->get('useComposerAsArtistGenres'))) if !$genreMatch;
```

Comment:
You could `break` out of the loop once yo have a match. And/or you could use a conditional assignment `$genreMatch ||= grep ...` instead of the appended condition. And as there's no need for a client for this kind of request, you could leave that part out, too.

But that's cosmetics. I doubt any of this will make a considerable difference 😀.